sdk/provisioning/Azure.Provisioning/src/Infrastructure.cs

```diff
@@ -36,34 +38,184 @@ public void Build(string? outputPath = null)
+        {
+            ModuleConstruct? construct = null;
+            if (NeedsModuleConstruct(resource, resourceTree))
+            {
+                construct = new ModuleConstruct(resource);
+
+                if (parentScope == null)
+                {
+                    root = construct;
+                    construct.IsRoot = true;
+                }
+            }
+
+            if (construct != null)
+            {
+                parentScope?.AddConstruct(construct);
+            }
+
+            if (parentScope != null)
+            {
+                parentScope.AddResource(resource);
+                resource.ModuleScope = parentScope;
+            }
+
+            parentScope = parentScope ?? construct;
+            if (parentScope != null)
+            {
+                foreach (var parameter in resource.Parameters)
+                {
+                    parentScope.AddParameter(parameter);
+                }
+
+                foreach (var parameterOverrideDictionary in resource.ParameterOverrides)
+                {
+                    foreach (var parameterOverride in parameterOverrideDictionary.Value)
+                    {
+                        parameterOverride.Value.Source = parentScope;
+                        parameterOverride.Value.Value = GetParameterValue(parameterOverride.Value, parentScope);
+                    }
+                }
+
+                foreach (var output in resource.Outputs)
+                {
+                    var moduleOutput = new Output(output.Name, output.Value, parentScope, output.IsLiteral, output.IsSecure);
+                    parentScope.AddOutput(moduleOutput);
+                }
+            }
+
+            foreach (var child in resourceTree[resource])
+            {
+                BuildModuleConstructs(child, resourceTree, construct ?? parentScope, ref root);
+            }
+        }
+
+        private static string GetParameterValue(Parameter parameter, IConstruct scope)
+        {
+            // If the parameter is a parameter of the module scope, use the parameter name.
+            if (scope.GetParameters(false).Contains(parameter))
+            {
+                return parameter.Name;
+            }
+            // Otherwise we assume it is an output from the current module.
+            if ( parameter.Source is null || ReferenceEquals(parameter.Source, scope))
+            {
+                return parameter.Value!;
+            }
+
+            return $"{parameter.Source.Name}.outputs.{parameter.Name}";
+        }
+
+        private static ConstructScope ResourceToConstructScope(Resource resource)
+        {
+            return resource switch
+            {
+                Tenant => ConstructScope.Tenant,
+                ResourceManager.Subscription => ConstructScope.Subscription,
+                //TODO managementgroup support
```

Comment:
can we open a tracking issue in our project and link it here

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/42036